app/src/main/kotlin/pro/qyoga/app/therapist/appointments/core/schedule/CalendarPageModel.kt

```diff
@@ -83,8 +87,26 @@ data class CalendarPageModel(
             return timeMarkRows
         }
 
-        private fun Appointment.fallsIn(wallClockDateTime: LocalDateTime, span: Duration) =
-            this.wallClockDateTime >= wallClockDateTime && this.wallClockDateTime < wallClockDateTime + span
+        private fun Appointment.calendarDateTime(currentUserTimeZone: ZoneId) =
+                this.dateTime.atZone(currentUserTimeZone).toLocalDateTime()
+
+        private fun Appointment.endCalendarDateTime(currentUserTimeZone: ZoneId) =
+                (this.dateTime + this.duration).atZone(currentUserTimeZone).toLocalDateTime()
+
+        private fun Appointment.fallsIn(wallClockDateTime: LocalDateTime, span: Duration, currentUserTimeZone: ZoneId) =
+                (this.calendarDateTime(currentUserTimeZone) >= wallClockDateTime
+                        && this.calendarDateTime(currentUserTimeZone) < wallClockDateTime + span)
+                        || (wallClockDateTime.hour == 0 && wallClockDateTime.minute == 0
+                        && this.calendarDateTime(currentUserTimeZone) < wallClockDateTime
+                        && this.endCalendarDateTime(currentUserTimeZone) >= wallClockDateTime)
```

Comment:
Давай так:
```suggestion
        private fun Appointment.fallsIn(wallClockDateTime: LocalDateTime, span: Duration, currentUserTimeZone: ZoneId) = if (wallClockDateTime.isDayBorder) {
        this.calendarDateTime(currentUserTimeZone) < wallClockDateTime
                        && this.endCalendarDateTime(currentUserTimeZone) >= wallClockDateTime
    } else {
        this.calendarDateTime(currentUserTimeZone) >= wallClockDateTime
                        && this.calendarDateTime(currentUserTimeZone) < wallClockDateTime + span
    }

private val LocalDateTime.isDayBorder() = 
    hour == 0 && minute == 0
```

Это сделает очевидным то, что у нас есть два случая.

Ещё, если есть желание, можно поиграться с таким синтаксисом:
```kotlin
 wallClockDateTime in this.calendarDateTime(currentUserTimeZone)..this.endCalendarDateTime(
                currentUserTimeZone
```